pkg/pool-manager/mac-entry.go

```diff
@@ -0,0 +1,38 @@
+package pool_manager
+
+import "time"
+
+func (m macEntry) isPending() bool {
+	if m.transactionTimestamp != nil {
+		return true
+	}
+	return false
+}
+
+func (m macEntry) addTime(duration time.Duration) time.Time {
+	return m.transactionTimestamp.Add(duration)
+}
+
+func (m macEntry) resetTimestamp() macEntry {
+	return macEntry{
+		instanceName:         m.instanceName,
+		macInstanceKey:       m.macInstanceKey,
+		transactionTimestamp: nil,
+	}
+}
+
+func (m macEntry) setTimestamp(timestamp *time.Time) macEntry {
+	return macEntry{
+		instanceName:         m.instanceName,
+		macInstanceKey:       m.macInstanceKey,
+		transactionTimestamp: timestamp,
+	}
+}
+
+func (m macEntry) createOrUpdate(instanceName, macInstanceKey string, timestamp *time.Time) macEntry {
```

Comment:
this does not make sense you are creagin at macEntry from a function that receive a macEntry, this kind of constructors are not needed

Reply:
I agree